### PR TITLE
testing: improve expectEqualSlices to print strings

### PR DESCRIPTION
**Pull request by vrischmann**
This PR prints `expected` and `actual` as strings if they're both `[]const u8` and valid UTF-8.

I took a shot at implementing a suggestion from the TODO but perhaps it would make more sense to have a dedicated function `expectString` or something.

### CODE REVIEW

**Comment by squeek502** [comment]
This will only show up when the lengths match but the contents don't match? That's probably a slightly rare case when dealing with strings, right?

**Comment by vrischmann** [comment]
You're right, I'll fix that

File: lib/std/testing.zig [hunk 1]
```diff
@@ -172,11 +172,24 @@ test "expectEqual.union(enum)" {
 pub fn expectEqualSlices(comptime T: type, expected: []const T, actual: []const T) void {
     // TODO better printing of the difference
     // If the arrays are small enough we could print the whole thing
-    // If the child type is u8 and no weird bytes, we could print it as strings
     // Even for the length difference, it would be useful to see the values of the slices probably.
+    if ((@typeInfo(T) == .Int) and @typeInfo(T).Int.bits == 8) {
+        const expected_string = @as([]const u8, expected);
+        const actual_string = @as([]const u8, actual);
+
+        if (std.unicode.utf8ValidateSlice(expected_string) and std.unicode.utf8ValidateSlice(actual_string)) {
```
Comment (data-man) [inline lib/std/testing.zig]: Please no. Slices can be just bytes.

**Comment by BarabasGitHub** [comment]
Sorry, but I really dislike this. I don't want my data that happens to be `u8` to be printed as utf8. Especially utf8 is bad, because it can combine multiple bytes into one character. 

Just make an `expectEqualUtf8` or something if you need this.

**Comment by vrischmann** [comment]
I can do that, I didn't know to which extent the API can be changed. It would be easier to use too.

**Comment by BarabasGitHub** [comment]
@vrischmann I'm no authority on Zig though. Just so you know.

**Comment by vrischmann** [comment]
Thanks for clarifying. This PR can still be reopened if necessary but I agree wih you that a dedicated function would be better anyway.

**Pull request closed without merging**